onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -616,7 +616,7 @@ static void RegisterExecutionProviders(InferenceSession* sess, const std::vector
           sess, *onnxruntime::CreateExecutionProviderFactory_ArmNN(sess->GetSessionOptions().enable_cpu_mem_arena));
 #endif
     } else if (type == kDmlExecutionProvider) {
-#ifdef USE_DML
+#if defined(USE_DML) && !defined(USE_CUDA)
```

Comment:
should we log a warning message saying DML was not registered for so and so reason? Otherwise if a package is built with a certain execution provider it is expected to be registered when using python packages right

Reply:
Also do you know how common is it to encounter such a scenario where both DML and CUDA are enabled together? In the current solution there is no way to use DML EP since CUDA is registered by default...

Reply:
If someone needs to use DML, he/she should build DML without CUDA.  Then it will work.

I feel it would be very common when memcpy node is involved.

Reply:
well yes they can build DML without CUDA so may be we just log a warning here to let users know DML was not registered because DML and CUDA cannot be enabled simultaneously

Reply:
Then probably we can totally disable such build combinations until it get fixed?  Nobody plans to ship DML and CUDA together.

Reply:
After splitting them, DML crashed. :-( 
I will continue working on this next week.

Reply:
https://github.com/microsoft/onnxruntime/pull/7987 is merged now.

Reply:
Thanks.